tensorflow/python/distribute/mirrored_strategy.py

```diff
@@ -331,8 +335,12 @@ def _initialize_strategy(self, devices):
   def _initialize_single_worker(self, devices):
     """Initializes the object for single-worker training."""
     self._devices = tuple(device_util.canonicalize(d) for d in devices)
-    self._input_workers = input_lib.InputWorkers(
-        ((device_util.canonicalize("/device:CPU:0", devices[0]), devices),))
+    if self._replication_mode == distribute_lib.InputReplicationMode.PER_WORKER:
```

Comment:
i am a little confused as to which changes in input_lib are still needed after changes in this file to adjust the input workers. 

let's add tests for these changes.

Reply:
Please check added test for input_lib_test.